### PR TITLE
[FSDP Optim State] Remove checkpoint prefix

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80480

Remove `_checkpoint_wrapped_module` prefixes when creating keys for optimizer state_dict.

Having these does not actually create an issue for optim_state_dict save / load, but we'd like to strip these keys out for downstream code that consumes these APIs typically expecting checkpointing prefixes to not exist (as checkpointing should be a transparent operation which should not change module / parameter names).

### CODE REVIEW

File: test/distributed/fsdp/test_fsdp_optim_state.py [hunk 1]
```diff
@@ -477,12 +480,21 @@ def test_full_optim_state_dict_keys(self):
...
+        )
         optim = torch.optim.Adam(wrapped_model.parameters(), lr=1e-3)
         self._step_model(model, optim, device)
         optim_state_dict = FSDP.full_optim_state_dict(wrapped_model, optim, rank0_only=False)
         with FSDP.state_dict_type(wrapped_model, StateDictType.FULL_STATE_DICT):
             state_dict = wrapped_model.state_dict()
         self.assertEqual(optim_state_dict["state"].keys(), state_dict.keys())
+        # Check that checkpointing prefix was indeed stripped.
+        for key in optim_state_dict["state"].keys():
+            self.assertFalse(_CHECKPOINT_PREFIX in key)
```
Comment (awgu) [inline test/distributed/fsdp/test_fsdp_optim_state.py:497]: nit: 😄 
```suggestion
            self.assertNotIn(_CHECKPOINT_PREFIX, key)
```
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotIn

**Review by awgu — APPROVED** [review]
Thanks for the super fast fix. This looks good to me!

File: test/distributed/fsdp/test_fsdp_optim_state.py [hunk 2]
```diff
@@ -477,12 +480,21 @@ def test_full_optim_state_dict_keys(self):
...
+            check_fn=lambda module: isinstance(module, torch.nn.Sequential)
+        )
         optim = torch.optim.Adam(wrapped_model.parameters(), lr=1e-3)
         self._step_model(model, optim, device)
         optim_state_dict = FSDP.full_optim_state_dict(wrapped_model, optim, rank0_only=False)
         with FSDP.state_dict_type(wrapped_model, StateDictType.FULL_STATE_DICT):
             state_dict = wrapped_model.state_dict()
         self.assertEqual(optim_state_dict["state"].keys(), state_dict.keys())
+        # Check that checkpointing prefix was indeed stripped.
+        for key in optim_state_dict["state"].keys():
```
Comment (awgu) [inline test/distributed/fsdp/test_fsdp_optim_state.py:496]: one more nit:
```suggestion
        for key in optim_state_dict["state"]:
```

**Review by fegin — APPROVED** [review]
LGTM

**Comment by rohan-varma** [comment]
CI failures are unrelated

**Comment by rohan-varma** [comment]
@suo my bad I guess the PR is somehow causing the issue in https://hud.pytorch.org/pytorch/pytorch/commit/04c50fec1ceec5b2d78934921f1cb366ff3fac81, it is very strange since none of those failing APIs are touched in this PR.

**Pull request closed without merging**